Emit histogram of number of lookup table items sent in each restore

dannyroberts opened the pull request:
both total and split out by global vs user

## Summary
A beginning step in https://dimagi-dev.atlassian.net/browse/SAAS-12010. Will be particularly helpful in picking the details of how we want to limit lookup table sizes (Items per lookup table? Total number of items? Total number of global items and user items separately?)

## Feature Flag
None

## Product Description
None

## Safety Assurance

- [x] Risk label is set correctly
- [x] All migrations are backwards compatible and won't block deploy
- [x] The set of people pinged as reviewers is appropriate for the level of risk of the change
- [x] If QA is part of the safety story, the "Awaiting QA" label is used
- [x] I have confidence that this PR will not introduce a regression for the reasons below

### Automated test coverage

https://github.com/dimagi/commcare-hq/blob/38102e873be50289efa9b7e991ef972df538d33a/corehq/apps/fixtures/tests/test_fixture_data.py#L34 covers this code over a number of code paths.

### QA Plan

Check the metric being generated on staging (and also thus that the code path doesn't break in some way surprisingly not covered by the test). (No need to involve QA team.)

### Safety story
This is a pretty small change (ostensibly just adding metrics), fairly well covered by tests, and I'm going to do a sanity check on staging before rolling it out. 

### Rollback instructions

<!--
If this PR follows standards of revertability, check the box below.
Otherwise replace it with detailed instructions or reasons a rollback is impossible.
-->

- [x] This PR can be reverted after deploy with no further considerations

dannyroberts committed:
Emit histogram of number of lookup table items sent in each restore

both total and split out by global vs user

snopoke commented on corehq/apps/fixtures/fixturegenerators.py:
```diff
@@ -61,11 +60,29 @@ def __call__(self, restore_state):
+                buckets=[1, 100, 1000, 10000, 30000, 100000, 300000, 1000000],
+                bucket_unit='',
+                tags={
+                    'domain': restore_user.domain
```
are you sure you want to tag all domains?

dannyroberts replied:
I think there's value in it? An alternative is to tag only on restores with more than 10,000 lookup table items, but then I might not be able to get a good impression of the full distribution for a given domain, just the distribution of their more lookup-table-intensive syncs. For example if we roll out a caching change, we won't be able to see the jump in a particular domain's number of restores with less than 1 (i.e. zero) lookup table items.

Any thoughts there? I suppose we could also remove half of the buckets and cut the metric count in half. Or require at least one sync with more than 30k lookup table items for a domain to be considered a "domain of interest" for this metric and then always report domains of interest, but that's getting a bit more complicated.

snopoke reviewed (commented):
Not blocking but these values are mostly going to be constant for a domain. Is there a reason we need to see this trend over time? Would a once off analysis like https://docs.google.com/spreadsheets/d/1xCA2JRKgecrxZ_VGtafoU57rxXHBh4fqsuBPaVHvZGs/edit#gid=332485977 not give us enough information?

snopoke reviewed (approved):
@dannyroberts that's all fine. I was just checking that you though there was enough value in this.

dannyroberts commented on corehq/apps/fixtures/fixturegenerators.py:
```diff
@@ -61,11 +60,29 @@ def __call__(self, restore_state):
             else:
                 user_types[data_type._id] = data_type
-        items = []
+        items = global_items = user_items = []
```
Bad bad bad. Very bad. Glad I put this on staging.

dannyroberts replied:
OK I over-racted. This was a bug, but it only affected metrics. Below global_items and user_items are reassigned if they're not empty, and otherwise they're not used except for metrics. That's why tests didn't catch this.

dannyroberts reviewed (commented):
Found bug, will update

dannyroberts committed:
Fix mistake in fixture count metrics

Three variables were getting set to the same empty list
instead of to different empty lists as intended.
This appears to have only affected the metric values themselves
incidentally, which is why tests didn't catch it.

dannyroberts commented:
I feel like I should update a test here. I'm a little surprised tests didn't catch it, but I think it's pretty clear what to test for that would have caught that error.

dannyroberts commented:
OK I tested the fix to the metrics on staging, and it's reporting the right thing now. I also took a bit of time to confirm that these lines are well-tested by changing the output in important ways and making tests fail. The things that make the test fail include: double-adding global_items to items; double-adding user_items to items; reversing items; changing the order in which global_items and user_items are added. All of those break the test, so with those things combined I'm pretty confident that this PR won't introduce a fixture bug. I had a little scare because of my metrics bug, but it does seem that even that would not have caused any user-facing bugs.

dannyroberts commented:
I realized today that global_items and user_items actually return a list of _xml-formatted strings_ (or something like that), not a flat list of items. So this metric is effectively bucketing by lookup table count rather than lookup table _item_ count. I'll look into how to fix that
